src/app/OperationalSessionSetup.cpp

```diff
@@ -423,6 +434,50 @@ CHIP_ERROR OperationalSessionSetup::LookupPeerAddress()
     return Resolver::Instance().LookupNode(request, mAddressLookupHandle);
 }
 
+void OperationalSessionSetup::PerformLookupOnExistingSession()
+{
+    bool isConnected = false;
+
+    if (mPerformingLookupOnConnectedSession)
+    {
+        // We are already in the middle of a lookup from a previous call to
+        // PerformLookupOnExistingSession. In that case we will just exit right away as
+        // we are already looking to update the results from the previous lookup.
+        return;
+    }
+
+    if (mState == State::Uninitialized)
+    {
+        // Thing are very wrong and we are just going to call DequeueConnectionCallbacks which will
+        // release ourself.
+        DequeueConnectionCallbacks(CHIP_ERROR_INCORRECT_STATE);
+        // Do not touch `this` instance anymore; it has been destroyed in DequeueConnectionCallbacks.
+        // While it is odd to have an explicit return here at the end of the function, we do so
+        // as a precaution in case someone later on adds something to the end of this function.
+        return;
+    }
+
+    if (mState == State::NeedsAddress)
+    {
+        // When mState is State::NeedsAddress this means that OperationalSessionSetup has been
+        // freshly created and we are expecting there to be an already established session
+        // that needs to have LookupPeerAddress.
+        isConnected = AttachToExistingSecureSession();
```

Comment:
A session could be defunct (and hence AttachToExistingSecureSession would not find it), but we're trying to send on it (from an existing exchange) and the re-lookup should update the address as needed...

Maybe what we should consider doing is this:
1. OperationalSessionSetup has an mUpdatingAddressForExistingSessions boolean as in this PR.
2. FindExistingSessionSetup allows specifying whether we want ones with that boolean true or false.
3. We never intermix an actual session establishment and an address update.  We just use separate OperationalSessionSetup objects for them.  Address updates can get coalesced with other address updates but not with actual session establishment, and vice versa.
4. The address update case goes through the normal state machine (NeedsAddress, ResolvingAddress) but in `UpdateDeviceData` skips the `EstablishConnection` bits if mUpdatingAddressForExistingSessions is false.
5. We can consider doing `UpdateAllSessionsPeerAddress` no matter what in UpdateDeviceData...

Thoughts?